### DISCUSSION TITLE
Understanding unsupervised loss in GraphSAGE

### DISCUSSION THREAD

**Discussion by RituPande** [danielegrattarola/spektral#320] [Q&A] [answered]
I have a slight problem understanding the unsupervised loss function given in paper "Inductive Representation Learning on Large Graphs" pg.5 equation 1

J (z<sub>u</sub> ) = - log (&sigma; (z<sub>u</sub><sup> T</sup>  z<sub>v</sub> )) -Q E <sub> v<sub>n</sub> ~P <sub>n</sub>(v)</sub> log (&sigma; (-z<sub>u</sub><sup> T</sup>  z<sub>v</sub> )) 

I understand that the first half of the equation is trying to minimize the loss for nodes close to each other in random walk and the second half is trying to minimize the loss for negative samples. However, I think the symbols used in the equation confuse me a little. 
1.  It is not clear to me if  z<sub>v </sub> in the first half of the equation is for all v's ( positive samples) or only a single instance of v.
2.  Does the right part  of the equation require taking average of log (&sigma; (-z<sub>u</sub><sup> T</sup>  z<sub>v</sub> )) of all negative samples and then multiply the average with  the number (Q) of negative samples taken ?
3. If my understanding of point-2 is correct, I am not sure of the rationale multiplying the loss of negative samples with the the number of samples rather than proportion of positive samples to balance the loss in case it is unbalanced b/w positive and negative samples.

**Comment by RituPande** [comment, marked as answer]
I understand the equation now after studying Noise Contrastive Estimation.

**Reply by danielegrattarola** [reply]
Hi, sorry I've been offline for the past few days. 
Good that you solved it on your own :D

Cheers
